### PR TITLE
convert macros to have command associations

### PR DESCRIPTION
**Pull request by grosser**
Get rid of command column in preparation for macros/stage unification ... see https://github.com/zendesk/samson/pull/769

existing `command` column will get converted into a actual `Command` new UI:

screen shot 2016-04-07 at 12 01 01 pm

 - includes a bit of duplication, but that will go away once Macros and Stage are merged
 - migration is using only clean classes to not break once macro becomes a Stage subclass

@zendesk/samson 

### Risks
- Level: Low/Med/High

### CODE REVIEW

**Comment by jonmoter** [comment]
Can you fix the CodeClimate issue?

**Comment by grosser** [comment]
extracted into a module ... good to go ?

File: app/models/concerns/has_commands.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module HasCommands
+  def self.included(base)
+    base.class_eval do
+      before_save :build_new_project_command
+      attr_writer :command
+    end
+  end
+
+  def command
```
Comment (jonmoter) [inline app/models/concerns/has_commands.rb]: How about `script` instead?

File: app/models/concerns/has_commands.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+module HasCommands
```
Comment (jonmoter) [inline app/models/concerns/has_commands.rb]: Add a comment that it expects `command_associations` to be defined by including class.

**Comment by jonmoter** [comment]
Okay, looks good. :+1:

**Comment by grosser** [comment]
added comment + made migration reversible

**Pull request merged**